### PR TITLE
FIX: Send additional headers to replication to AWS

### PR DESCRIPTION
**Pull request by bennettbuchanan**

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
E2E failed due to flaky Utapi test:
```
Utapi tests Utapi: USERS:: Basic UserPolicy BasicUserPolicy:: Attaching a policy to Bart that grant him access to getMetrics should allow Bart to access bucketMetrics
```

Rerunning now.

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
Got retried here http://ci.ironmann.io/gh/scality/Integration/17000

File: extensions/replication/utils/QueueEntry.js [hunk 1]
```diff
@@ -73,6 +73,20 @@ class QueueEntry extends ObjectMD {
...
 
+    getUserMetadata() {
+        const metaHeaders = {};
+        const data = this.getValue();
+        Object.keys(data).forEach(key => {
+            if (key.startsWith('x-amz-meta-')) {
+                metaHeaders[key] = data[key];
+            }
+        });
+        if (metaHeaders !== {}) {
```
Comment (rahulreddy) [inline extensions/replication/utils/QueueEntry.js]: This conditional will always be true.

```js
> x = {}
{}
> x === {}
false
> x !== {}
true
```
Best thing is to use `Object.keys.length > 0`

Comment (bennettbuchanan) [reply extensions/replication/utils/QueueEntry.js]: Updated to `if (Object.keys(metaHeaders).length > 0)`

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
E2E failed due to what appears to be a flaky multiple backend test, as this particular test was observed previously to both pass and fail on the same branch.

Rerunning here: http://ci.ironmann.io/gh/scality/Integration/17007.

**Comment by bennettbuchanan** [comment]
E2E passed on the retry here: http://ci.ironmann.io/gh/scality/Integration/17007

**Pull request merged**